wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt

```diff
@@ -358,17 +366,46 @@ class MainViewModel @Inject constructor(
     fun addFavoriteEntity(entityId: String) {
         viewModelScope.launch {
             favoritesDao.addToEnd(entityId)
+            addCachedFavorite(entityId)
         }
     }
 
     fun removeFavoriteEntity(entityId: String) {
         viewModelScope.launch {
             favoritesDao.delete(entityId)
+            removeCachedFavorite(entityId)
+        }
+    }
+
+    fun getCachedEntity(entityId: String): FavoriteCaches =
+        FavoriteCachesDao.get(entityId)
```

Comment:
1. This might return `null` if there is no favorite.

2. This is called directly from the composable function, so please implement some form of caching so that the app isn't constantly reading from the database. For example implement a `getAll()` function in the DAO that is used when the viewmodel is initialized, and return values from that list.